src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -150,7 +150,7 @@ public bool IsAnonymousFunction(SyntaxNode node)
         }
 
         public bool IsNamedParameter(SyntaxNode node)
-            => node.CheckParent<NameColonSyntax>(p => p.Name == node);
+            => node.CheckParent<NameColonSyntax>(p => p.Name == node) || (node is ArgumentSyntax arg && arg.NameColon != null);
```

Comment:
something is wonky here.  the old logic was just busted.  we should just remove that.  And if something breaks it should then call a new method called `IsNameOfNamedArgument`

we absolutely should not have both types of logic running together.

Reply:
I've split it as you suggested. :)